sdk/identity/identity/src/msal/utils.ts

```diff
@@ -149,24 +149,25 @@ export class MsalBaseUtilities {
    * Handles MSAL errors.
    */
   protected handleError(scopes: string[], error: Error, getTokenOptions?: GetTokenOptions): Error {
-    if (error instanceof msalCommon.AuthError) {
-      switch (error.errorCode) {
+    if (error.name === "AuthError" || error.name === "ClientAuthError") {
```

Comment:
Is ClientAuthError a separate error class? If so, then am guessing that the previous instanceOf check did not work for this case?

Reply:
ClientAuthError inherits AuthError. The previous one worked for both.

Reply:
Do we have confirmation from the msal team that there are no other error classes that inherit AuthError?

Reply:
These are all the instances of things that extend AuthError:

```
./lib/msal-browser/src/error/BrowserAuthError.ts:149:export class BrowserAuthError extends AuthError {
./lib/msal-browser/src/error/BrowserConfigurationAuthError.ts:47:export class BrowserConfigurationAuthError extends AuthError {
./lib/msal-common/src/error/ServerError.ts:11:export class ServerError extends AuthError {
./lib/msal-common/src/error/ClientAuthError.ts:179:export class ClientAuthError extends AuthError {
./lib/msal-core/src/error/ServerError.ts:21:export class ServerError extends AuthError {
./lib/msal-core/src/error/ClientAuthError.ts:96:export class ClientAuthError extends AuthError {
```

I can see some improvements I can do. I'll push another commit.

Reply:
I have pushed this commit that considers BrowserAuthError and BrowserConfigurationAuthError! Good catch! Thank you! https://github.com/Azure/azure-sdk-for-js/pull/14763/commits/d67b6453c346b2c8806970a987795b17646899ef

Reply:
One concern here is that if msal-common adds new error classes that keep extending from AuthError, we will have to keep playing catch up. Just something to keep in mind I guess.